stor/s3.py

```diff
@@ -765,3 +769,34 @@ def to_url(self):
         """Returns HTTP url for object (virtual host-style)"""
         return u'https://{bucket}.s3.amazonaws.com/{key}'.format(bucket=self.bucket,
                                                                  key=self.resource)
+
+    def restore(self, tier='Bulk', days=10):
+        """Issue a restore command for a single object from glacier.
+
+        Args:
+            tier (str, optional): restore speed (see Glacier docs for details)
+            days (int, optional): number of days to keep data in S3 post-restore.
+        Returns:
+            dict or None: restore response from S3 client
+        Note:
+            Calling ``restore()`` on a directory will not work correctly. Only use this for single
```

Comment:
can we detect this case and raise error?

Reply:
not really (aside from calling list() every single time you call restore which is both slow and costly) and tbh trying to do it just makes the API less useful for users (because either you try to string match to see things that look "prefix-like" and you block them or you miss cases where you should've seen it).

Right now this will raise an NotFoundError, like this:

```
In [2]: stor.Path('s3://some-bucket/my/key').restore()
---------------------------------------------------------------------------
NotFoundError                             Traceback (most recent call last)
<ipython-input-2-4275e295256d> in <module>()

/Users/jtratner/stor/stor/s3.py in restore(self, tier, days)
    794                                  Key=self.resource,
    795                                  RestoreRequest={'Days': days,
--> 796                                                  'GlacierJobParameters': {'Tier': tier}})
    797         except exceptions.RestoreAlreadyInProgressError:
    798             logger.debug('restore already started, not doing anything')

/Users/jtratner/stor/stor/s3.py in _s3_client_call(self, method_name, *args, **kwargs)
    200             return method(*args, **kwargs)
    201         except botocore_exceptions.ClientError as e:
--> 202             six.raise_from(_parse_s3_error(e, **kwargs), e)
    203
    204     def _get_s3_iterator(self, method_name, *args, **kwargs):

/usr/local/lib/python2.7/site-packages/six.pyc in raise_from(value, from_value)
    735 else:
    736     def raise_from(value, from_value):
--> 737         raise value
    738
    739

NotFoundError: The specified key does not exist. Bucket: some-bucket Key: my/key
```

I think that's good enough - we could potentially try to catch the `NotFoundError` and append ` - is it a prefix/directory? If so call restore() on each object individually` to the exception message

Reply:
What you have is good enough. 👍